### ISSUE TITLE
IME issue in chat and F1

### ISSUE THREAD

**Issue by Learath2** [ddnet/ddnet#4888]
Type something in either Korean or Japanese, use backspace to delete it and the issue becomes obvious. Also happens after you send one message in japanese, then re-open chat. It seems a buffer is not getting cleaned up. Same issue does not seem to happen in Menus.

![image](https://user-images.githubusercontent.com/490500/159925358-725af88d-5823-43d9-93f8-f0771db0475c.png)

**Comment by C0D3D3V** [comment]
it's strange that this only happens with japanese and korean, but not with chinese.

**Comment by C0D3D3V** [comment]
i just tested it with japanese too, it doesn't happen for me

**Comment by def-** [comment]
Can you bisect if this was introduced recently? I think we had some str_* changes, maybe one of them broke it.

**Comment by def-** [comment]
![Screenshot 2022-03-31 at 12 02 20](https://user-images.githubusercontent.com/2335377/161030241-41edf0da-b1a5-4fe7-9427-a1e180071504.png)
I think I can reproduce this on macOS, will try to bisect.

**Comment by def-** [comment]
Sigh, my fault:
```
ded32c43344352985ea40806ed9ecff4212514d4 is the first bad commit
commit ded32c43344352985ea40806ed9ecff4212514d4
Author: Dennis Felsing <dennis@felsin9.de>
Date:   Fri Feb 18 12:21:48 2022 +0100

    Fix MinGW warnings (fixes #4731)

 src/base/system.cpp                    | 10 +++++-----
 src/engine/client/backend_sdl.cpp      |  2 +-
 src/engine/server/databases/sqlite.cpp |  6 +++++-
 3 files changed, 11 insertions(+), 7 deletions(-)
```
